modules/swagger-codegen/src/main/resources/rust/api.mustache

```diff
@@ -0,0 +1,96 @@
+{{>partial_header}}
+use std::rc::Rc;
+use std::borrow::Borrow;
+
+use hyper;
+use serde_json;
+use futures;
+use futures::{Future, Stream};
+
+use super::{Error, configuration};
+
+pub struct {{{classname}}}Impl<C: hyper::client::Connect> {
+    configuration: Rc<configuration::Configuration<C>>,
+}
+
+impl<C: hyper::client::Connect> {{{classname}}}Impl<C> {
```

Comment:
FWIW I don't see a lot of `Impl` suffixes in Rust. I believe it would be more idiomatic to make this something like `{{{classname}}}Connection` or rename the trait to `{{{classname}}}Client`.